File: sop-sdk/sdk-csharp/SDKCSharp/Response/BaseResponse.cs
using System;
using Newtonsoft.Json;

namespace SDKCSharp.Response
{
    /// <summary>
    /// 返回的Response，新建Response要继承这个类
    /// </summary>
    public abstract class BaseResponse
    {
        
        /// <summary>
        /// 状态码，0表示成功，其它都是失败
        /// </summary>
        [JsonProperty("code")]
        public string Code { get; set; }

        /// <summary>
        /// 消息，如果有错误则为错误信息
        /// </summary>
        [JsonProperty("msg")]
        public string Msg { get; set; }

        /// <summary>
        /// 错误状态码
        /// </summary>
        /// <value>The sub code.</value>
        [JsonProperty("sub_code")]
        public string SubCode { get; set; }

        /// <summary>
        /// 错误消息
        /// </summary>
        /// <value>The sub message.</value>
        [JsonProperty("sub_msg")]
        public string SubMsg { get; set; }

        /// <summary>
        /// 响应原始内容
        /// </summary>
        [JsonIgnore]
        public string Body { get; set; }

        /// <summary>
        /// 是否成功
        /// </summary>
        /// <returns></returns>
        public bool IsSuccess()
        {
            return string.IsNullOrEmpty(SubCode);
        }
    }
}

File: sop-sdk/sdk-java/src/main/java/com/gitee/sop/sdk/response/BaseResponse.java
package com.gitee.sop.sdk.response;

import com.alibaba.fastjson.annotation.JSONField;
import com.gitee.sop.sdk.sign.StringUtils;
import lombok.Getter;
import lombok.Setter;

/**
 * 返回对象，后续返回对象都要继承这个类
 * {
 * "alipay_trade_close_response": {
 * "code": "20000",
 * "msg": "Service Currently Unavailable",
 * "sub_code": "isp.unknown-error",
 * "sub_msg": "系统繁忙"
 * },
 * "sign": "ERITJKEIJKJHKKKKKKKHJEREEEEEEEEEEE"
 * }
 * @author tanghc
 */
@Setter
@Getter
public abstract class BaseResponse {

    @JSONField(name = "request_id")
    private String requestId;
    private String code;
    private String msg;
    @JSONField(name = "sub_code")
    private String subCode;
    @JSONField(name = "sub_msg")
    private String subMsg;
    @JSONField(serialize = false)
    private String body;

    @JSONField(serialize = false)
    public boolean isSuccess() {
        return StringUtils.isEmpty(subCode);
    }


}
